src/machine-learning/machine-learning.ts

```diff
@@ -252,13 +295,73 @@ export class ModelOptions {
   public displayName?: string;
   public tags?: string[];
 
-  public tfliteModel?: { gcsTFLiteUri: string; };
+  public tfliteModel?: { gcsTfliteUri: string; };
+}
 
-  protected toJSON(forUpload?: boolean): object {
-    throw new Error('NotImplemented');
+async function convertOptionstoContent(
+    options: ModelOptions, forUpload?: boolean,
+    storageClient?: StorageClient): Promise<object> {
+  const modelContent = deepCopy(options);
+  if (forUpload && modelContent.tfliteModel?.gcsTfliteUri) {
+    if (!storageClient) {
+      throw new FirebaseMachineLearningError(
+        'invalid-argument',
+        'Must specify storage client if forUpload and gcs Uri are specified.',
+      );
+    }
+    modelContent.tfliteModel.gcsTfliteUri = await signUrl(modelContent.tfliteModel.gcsTfliteUri, storageClient!);
+  }
+  return modelContent;
+}
+
+async function signUrl(unsignedUrl: string, storageClient: StorageClient): Promise<string> {
+  const MINUTES = 60 * 1000; // A minute in milliseconds.
+  const URL_VALID_DURATION = 10 * MINUTES;
+
+  const gcsRegex = /^gs:\/\/([a-z0-9_.-]{3,63})\/(.+)$/;
+  const matches = gcsRegex.exec(unsignedUrl);
+  if (!matches) {
+    throw new FirebaseMachineLearningError(
+      'invalid-argument',
+      `Invalid unsigned url: ${unsignedUrl}`);
+  }
+  const bucketName = matches[1];
+  const blobName = matches[2];
+  const bucket = storageClient.bucket(bucketName);
+  const blob = bucket.file(blobName);
+
+  try {
+    const signedUrl = blob.getSignedUrl({
+      action: 'read',
+      expires: Date.now() + URL_VALID_DURATION,
+    }).then((x) => x[0]);
+    return signedUrl;
+  } catch (err) {
+    throw new FirebaseMachineLearningError(
+      'internal-error',
+      `Error during signing upload url: ${err.message}`,
+    );
   }
 }
 
 function extractModelId(resourceName: string): string {
   return resourceName.split('/').pop()!;
 }
+
+
+function handleOperation(op: OperationResponse): Model {
+  if (op.done) {
+    if (op.response) {
+      return new Model(op.response);
+    } else if (op.error) {
+      handleOperationError(op.error);
+    }
+  }
+  throw new FirebaseMachineLearningError(
```

Comment:
This will run whenever `op.done == false`. Is that what you wanted?

Reply:
Yes. Backend currently only returns operations with done=true and we wrote this SDK with that expectation.